### PR TITLE
feat: storage registry support

### PR DESCRIPTION
**Pull request by CassOnMars**
## Motivation

Adds support for Storage registry contract, plus L2 integration

## Change Summary

Describe the changes being made in 1-2 concise sentences.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [x] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR does not require changes to the [protocol](https://github.com/farcasterxyz/protocol)
- [x] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

---

## PR-Codex overview
This PR adds support for storage events and includes the following changes:

- Adds support for storage events in the protobufs
- Updates the server to handle rent registry events
- Updates the logger to include rent registry and storage admin registry events
- Adds CLI option for L2 RPC URL
- Adds methods to submit rent registry and storage admin registry events in the mock hub
- Updates the sync engine to use L2 events provider

> The following files were skipped due to too many changes: `apps/hubble/src/network/sync/syncEngine.ts`, `apps/hubble/src/rpc/adminServer.ts`, `apps/hubble/src/test/utils.ts`, `packages/core/src/factories.test.ts`, `apps/hubble/src/storage/db/storageRegistryEvent.test.ts`, `packages/core/src/factories.ts`, `apps/hubble/src/storage/engine/index.ts`, `apps/hubble/src/storage/stores/storageEventStore.ts`, `apps/hubble/src/storage/db/storageRegistryEvent.ts`, `apps/hubble/src/eth/l2EventsProvider.test.ts`, `packages/hub-nodejs/src/generated/rpc.ts`, `packages/hub-web/src/generated/request_response.ts`, `packages/hub-nodejs/src/generated/request_response.ts`, `packages/core/src/protobufs/generated/request_response.ts`, `apps/hubble/src/hubble.ts`, `apps/hubble/src/storage/stores/storageEventStore.test.ts`, `packages/hub-web/src/generated/hub_event.ts`, `packages/hub-nodejs/src/generated/hub_event.ts`, `packages/core/src/protobufs/generated/hub_event.ts`, `packages/hub-web/src/generated/storage_event.ts`, `packages/hub-nodejs/src/generated/storage_event.ts`, `packages/core/src/protobufs/generated/storage_event.ts`, `apps/hubble/src/eth/l2EventsProvider.ts`, `packages/hub-web/src/generated/rpc.ts`, `apps/hubble/src/eth/abis.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: protobufs/schemas/storage_event.proto [hunk 1]
```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+enum StorageRegistryEventType {
+  STORAGE_REGISTRY_EVENT_TYPE_NONE = 0;
+  STORAGE_REGISTRY_EVENT_TYPE_RENT = 1;
+  STORAGE_REGISTRY_EVENT_TYPE_SET_PRICE = 2;
+  STORAGE_REGISTRY_EVENT_TYPE_SET_MAX_UNITS = 3;
+  STORAGE_REGISTRY_EVENT_TYPE_SET_DEPRECATION_TIMESTAMP = 4;
+  STORAGE_REGISTRY_EVENT_TYPE_SET_GRACE_PERIOD = 5;
```
Comment (sanjayprabhu) [inline protobufs/schemas/storage_event.proto:9]: Does the hub need to store/know about all these other event types? Wouldn't it only act on the RENT event?

File: apps/hubble/src/storage/stores/storageEventStore.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    return events;
+  }
+
+  /** Returns the events from the StorageAdminRegistry contract that affected the storage mechanics  */
+  async getStorageAdminRegistryEvents(): Promise<StorageAdminRegistryEvent[]> {
+    const iterator = await getStorageAdminRegistryEventsIterator(this._db);
+    const events: StorageAdminRegistryEvent[] = [];
+    let event: StorageAdminRegistryEvent | undefined;
+
+    while (iterator.isOpen) {
```
Comment (sanjayprabhu) [inline apps/hubble/src/storage/stores/storageEventStore.ts:67]: I believe we still need to call `.end()` see https://github.com/farcasterxyz/hub-monorepo/pull/1011#issue-1748622529? Might be better to use the `for await` pattern for fetching from iterators.

File: protobufs/schemas/storage_event.proto (hunk 1 above)
Comment (CassOnMars) [reply protobufs/schemas/storage_event.proto:9]: price is probably the only one that may not be important, but max units, deprecation timestamp, grace period are all relevant for pruning/informing hub operators about required capacity

File: apps/hubble/src/hubble.ts [hunk 3]
```diff
@@ -295,6 +331,14 @@ export class Hub implements HubInterface {
...
       );
     }
+
+    // if (this.l2RegistryProvider) {
+    //   this.updateRentRegistryEventExpiryJobWorker = new UpdateRentRegistryEventExpiryJobWorker(
+    //     this.updateRentRegistryEventExpiryJobQueue,
+    //     this.rocksDB,
+    //     this.l2RegistryProvider
+    //   );
+    // }
```
[the remark this answers is not in the document]
Comment (CassOnMars) [reply apps/hubble/src/hubble.ts:341]: commented out for indicating next insertion point in the followup support for pruning, safe to remove, but just helpful to keep in mind

**Pull request merged**